File: ideal-job-scheduler/ideal-job-scheduler-core/src/main/java/com/zzsong/job/scheduler/core/admin/repository/JobExecutorRepository.java
package com.zzsong.job.scheduler.core.admin.repository;

import com.zzsong.job.scheduler.core.admin.entity.JobExecutor;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.lang.Nullable;

import javax.annotation.Nonnull;

/**
 * @author 宋志宗
 * @date 2020/9/2
 */
public interface JobExecutorRepository
        extends JpaRepository<JobExecutor, Long>,
        JpaSpecificationExecutor<JobExecutor> {

    @Nullable
    JobExecutor findTopByAppName(@Nonnull String appName);
}

File: ideal-job-scheduler/ideal-job-scheduler-core/src/main/kotlin/cn/sh/ideal/job/scheduler/core/admin/repository/JobExecutorRepository.kt
package cn.sh.ideal.job.scheduler.core.admin.repository

import cn.sh.ideal.job.scheduler.core.admin.entity.JobExecutor
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor

/**
 * @author 宋志宗
 * @date 2020/8/20
 */
interface JobExecutorRepository : JpaRepository<JobExecutor, Long>, JpaSpecificationExecutor<JobExecutor> {

  fun findTopByAppName(appName: String): JobExecutor?
}